src/handlers/flow/FlowStreamHandler.cpp

```diff
@@ -40,17 +41,37 @@ static std::string ip_summarization(const std::string &val, SummaryData *summary
             if (!summary->asn_exclude_summary.empty() && std::any_of(summary->asn_exclude_summary.begin(), summary->asn_exclude_summary.end(), [&asn](const auto &prefix) {
                     return asn.size() >= prefix.size() && 0 == asn.compare(0, prefix.size(), prefix);
                 })) {
-                return val;
+                check_subnet = true;
             }
-            return asn;
-        } else if (summary->type == IpSummary::BySubnet) {
+            if (!check_subnet) {
+                return asn;
+            }
+        }
+        if (summary->type == IpSummary::BySubnet || check_subnet) {
             if (ipv4.isValid()) {
                 if (auto subnet = match_subnet(summary->ipv4_summary, ipv4.toInt()); subnet.has_value()) {
                     return subnet.value()->str;
+                } else if (summary->ipv4_wildcard.has_value()) {
+                    return pcpp::IPv4Address(ipv4.toInt() & (htobe32((0xFFFFFFFFu) << (32 - summary->ipv4_wildcard.value().cidr)))).toString()
```

Comment:
I would prefer to have this in a separate utility function - seems like something we should have in a common place.

Reply:
done